### PR TITLE
Improvements to comparisons

### PR DESCRIPTION
**Pull request by grimreaper**
We can avoid implementing our own comparison code.

### CODE REVIEW

File: code/src/java/pcgen/util/Comparators.java [hunk 1]
```diff
@@ -17,139 +17,66 @@
...
 	 * such as maintaining a sorted {@code JComboBoxEx} and the like.
 	 */
-	private static final class ToStringComparator implements Comparator<Object>, Serializable
+	private static final class ToStringComparator implements Comparator<Object>
 	{
-
 		@Override
 		public int compare(Object o1, Object o2)
 		{
-			// Treat null as the empty string.
```
Comment (thpr) [inline code/src/java/pcgen/util/Comparators.java]: I like this because it shows developer intent and is not the traditional way a compare would work

**Comment by LegacyKing** [comment]
@grimreaper Travis still failing.

**Comment by LegacyKing** [comment]
I restarted the Travis build in case the errors are fixed in another PR. For what it's worth, this LGTM.

**Comment by LegacyKing** [comment]
`> Task :compileJava
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/PCGenFrame.java:166: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
		ChooserFactory.setDelegate(this);
		^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/PCGenFrame.java:1647: warning: [removal] ChoiceHandler in pcgen.util.chooser has been deprecated and marked for removal
		Optional<ChoiceHandler> choiceHandler = ChooserFactory.getChoiceHandler();
		         ^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/PCGenFrame.java:1647: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
		Optional<ChoiceHandler> choiceHandler = ChooserFactory.getChoiceHandler();
		                                        ^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/sources/AdvancedSourceSelectionPanel.java:294: error: cannot find symbol
		wrap.sort(Comparators.toStringIgnoreCaseCollator());
		                     ^
  symbol:   method toStringIgnoreCaseCollator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/sources/SourceSelectionDialog.java:269: error: cannot find symbol
			ListFacade<SourceSelectionFacade> sources = new SortedListFacade<>(Comparators.toStringIgnoreCaseCollator(),
			                                                                              ^
  symbol:   method toStringIgnoreCaseCollator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/sources/SourceSelectionDialog.java:506: error: cannot find symbol
			ListFacade<SourceSelectionFacade> sources = new SortedListFacade<>(Comparators.toStringIgnoreCaseCollator(),
			                                                                              ^
  symbol:   method toStringIgnoreCaseCollator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/sources/SourceSelectionDialog.java:558: error: cannot find symbol
			}, defaultSelectedSource, Comparators.toStringIgnoreCaseCollator());
			                                     ^
  symbol:   method toStringIgnoreCaseCollator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/tabs/CompanionInfoTab.java:938: error: cannot find symbol
				int insertIndex = Collections.binarySearch(children, child, Comparators.toStringIgnoreCaseCollator());
				                                                                       ^
  symbol:   method toStringIgnoreCaseCollator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/tabs/CompanionInfoTab.java:973: error: cannot find symbol
				int index = Collections.binarySearch(children, child, Comparators.toStringIgnoreCaseCollator());
				                                                                 ^
  symbol:   method toStringIgnoreCaseCollator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/tabs/CompanionInfoTab.java:982: error: cannot find symbol
				children.sort(Comparators.toStringIgnoreCaseCollator());
				                         ^
  symbol:   method toStringIgnoreCaseCollator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/tabs/CompanionInfoTab.java:1027: error: cannot find symbol
				types.sort(Comparators.toStringIgnoreCaseCollator());
				                      ^
  symbol:   method toStringIgnoreCaseCollator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/tabs/CompanionInfoTab.java:1043: error: cannot find symbol
				int index = Collections.binarySearch(types, type, Comparators.toStringIgnoreCaseCollator());
				                                                             ^
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/tabs/CompanionInfoTab.java:1057: error: cannot find symbol
				int insertIndex = Collections.binarySearch(types, e.getKey(), Comparators.toStringIgnoreCaseCollator());
				                                                                         ^
  symbol:   method toStringIgnoreCaseCollator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/tabs/CompanionInfoTab.java:1111: error: cannot find symbol
				int index = Collections.binarySearch(types, type, Comparators.toStringIgnoreCaseCollator());
				                                                             ^
  symbol:   method toStringIgnoreCaseCollator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/tabs/ability/AbilityTreeTableModel.java:140: error: cannot find symbol
			this.cats = new SortedListFacade<>(Comparators.toStringIgnoreCaseComparator(), cats);
			                                              ^
  symbol:   method toStringIgnoreCaseComparator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/tabs/ability/AbilityTreeTableModel.java:189: error: cannot find symbol
			this.abilities = new SortedListFacade<>(Comparators.toStringIgnoreCaseComparator(),
			                                                   ^
  symbol:   method toStringIgnoreCaseComparator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/PCGenMenuBar.java:270: error: cannot find symbol
			setListModel(new SortedListFacade<>(Comparators.toStringIgnoreCaseCollator(), sources));
			                                               ^
  symbol:   method toStringIgnoreCaseCollator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/cdom/base/ConcreteTransitionChoice.java:221: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
			ChooserFactory.getDelegate().showGeneralChooser(chooserFacade);
			^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/util/chooser/RandomChooser.java:30: warning: [removal] ChoiceHandler in pcgen.util.chooser has been deprecated and marked for removal
public final class RandomChooser implements ChoiceHandler
                                            ^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/util/chooser/ChooserFactory.java:47: warning: [removal] ChoiceHandler in pcgen.util.chooser has been deprecated and marked for removal
	public static Optional<ChoiceHandler> getChoiceHandler()
	                       ^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/util/chooser/ChooserFactory.java:51: warning: [removal] RandomChooser in pcgen.util.chooser has been deprecated and marked for removal
			return Optional.of(new RandomChooser());
			                       ^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/npcgen/NPCGenerator.java:534: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
		ChooserFactory.useRandomChooser(); //$NON-NLS-1$
		^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/npcgen/NPCGenerator.java:782: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
			ChooserFactory.stopUsingRandomChooser();
			^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/Globals.java:976: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
		ChooserFactory.getDelegate().showGeneralChooser(chooserFacade);
		^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/analysis/StatApplication.java:118: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
			ChooserFactory.getDelegate().showGeneralChooser(chooserFacade);
			^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/analysis/ExchangeLevelApplication.java:89: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
						ChooserFactory.getDelegate().showGeneralChooser(chooserFacade);
						^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/analysis/SubClassApplication.java:108: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
			ChooserFactory.getDelegate().showGeneralChooser(chooserFacade);
			^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/analysis/SubClassApplication.java:116: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
				ChooserFactory.getDelegate().showGeneralChooser(chooserFacade);
				^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/analysis/SubClassApplication.java:176: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
				ChooserFactory.getDelegate().showGeneralChooser(chooserFacade);
				^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/analysis/EquipmentChoiceDriver.java:85: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
		ChooserFactory.getDelegate().showGeneralChooser(chooserFacade);
		^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/analysis/EquipmentChoiceDriver.java:133: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
					ChooserFactory.getDelegate().showGeneralChooser(chooserFacade);
					^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/analysis/SubstitutionClassApplication.java:65: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
		ChooserFactory.getDelegate().showGeneralChooser(chooserFacade);
		^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/chooser/CDOMChoiceManager.java:173: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
		ChooserFactory.getDelegate().showGeneralChooser(chooserFacade);
		^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/chooser/UserInputManager.java:68: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
		ChooserFactory.getDelegate().showGeneralChooser(chooserFacade);
		^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/kit/KitTemplate.java:101: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
			ChooserFactory.useRandomChooser(); //$NON-NLS-1$
			^
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/core/kit/KitTemplate.java:127: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
			ChooserFactory.stopUsingRandomChooser();
			^
/home/travis/build/PCGen/pcgen/code/src/java/plugin/lsttokens/deprecated/ModifyfeatchoiceToken.java:162: warning: [removal] ChooserFactory in pcgen.util.chooser has been deprecated and marked for removal
		ChooserFactory.getDelegate().showGeneralChooser(chooserFacade);
		^`

**Comment by karianna** [comment]
@grimreaper 

```
 symbol:   method toStringIgnoreCaseCollator()
  location: class Comparators
/home/travis/build/PCGen/pcgen/code/src/java/pcgen/gui2/tabs/ability/AbilityTreeTableModel.java:140: error: cannot find symbol
			this.cats = new SortedListFacade<>(Comparators.toStringIgnoreCaseComparator(), cats);
```

Is the actual compile error here

**Comment by grimreaper** [comment]
I have a better version of this somewhere.

**Pull request closed without merging**